ArcGISRuntimeSDKQt_CppSamples/Analysis/FindFeaturesUtilityNetwork/FindFeaturesUtilityNetwork.h

```diff
@@ -0,0 +1,112 @@
+// [WriteFile Name=FindFeaturesUtilityNetwork, Category=Analysis]
+// [Legal]
+// Copyright 2019 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifndef FINDFEATURESUTILITYNETWORK_H
+#define FINDFEATURESUTILITYNETWORK_H
+
+namespace Esri
+{
+namespace ArcGISRuntime
+{
+class Map;
+class MapQuickView;
+class UtilityNetwork;
+class UtilityElement;
+class UtilityTerminal;
+class UtilityTraceParameters;
+class FeatureLayer;
+class ServiceFeatureTable;
+class SimpleMarkerSymbol;
+class SimpleLineSymbol;
+class GraphicsOverlay;
+class ArcGISFeature;
+class IdentifyLayerResult;
+}
+}
+
+#include <QObject>
+#include <QUrl>
+#include <QUuid>
+#include "Point.h"
+
+class FindFeaturesUtilityNetwork : public QObject
+{
+  Q_OBJECT
+
+  Q_PROPERTY(Esri::ArcGISRuntime::MapQuickView* mapView READ mapView WRITE setMapView NOTIFY mapViewChanged)
+  Q_PROPERTY(bool terminalDialogVisisble MEMBER terminalDialogVisisble NOTIFY terminalDialogVisisbleChanged)
+  Q_PROPERTY(bool dialogVisible MEMBER dialogVisible NOTIFY dialogVisibleChanged)
+  Q_PROPERTY(QString dialogText MEMBER dialogText NOTIFY dialogTextChanged)
+  Q_PROPERTY(bool startingLocationsEnabled MEMBER startingLocationsEnabled NOTIFY startingLocationsChanged)
+  Q_PROPERTY(bool busy MEMBER m_busy NOTIFY busyChanged)
+
+public:
+  explicit FindFeaturesUtilityNetwork(QObject* parent = nullptr);
+  ~FindFeaturesUtilityNetwork();
+
+  static void init();
+
+  Q_INVOKABLE void multiTerminalIndex(int index);
+  Q_INVOKABLE void trace();
+  Q_INVOKABLE void reset();
+
+signals:
+  void mapViewChanged();
+  void terminalDialogVisisbleChanged();
+  void dialogVisibleChanged();
+  void dialogTextChanged();
+  void startingLocationsChanged();
+  void busyChanged();
+
+private slots:
+  void selectFeatures();
+  void selectUtilityElement(QUuid, const QList<Esri::ArcGISRuntime::IdentifyLayerResult*>& results);
```

Comment:
@DonMoJa I think the code itself is much more readable with these refactored into actual slots/methods. However, I think these should have more meaningful names. 

For example, you cannot tell what signal `selectFeatures` is reacting to. A potential improvement would be something like `void onTraceCompleted();`. That way it's clear to the reader that the slot executes when the  trace operation completes.

I would rename `selectUtilityElement` for the same reason.